streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBWindowStore.java

```diff
@@ -48,11 +49,12 @@
 
     private static final long USE_CURRENT_TIMESTAMP = -1L;
 
-    private static class Segment extends RocksDBStore<byte[], byte[]> {
+    // use the ByteBuffer wrapper for underlying rocksDB keys since they are used for hashing data structures
+    private static class Segment extends RocksDBStore<ByteBuffer, byte[]> {
```

Comment:
@joshng I was not concerning about performance, but rather the duplications of having `ByteArraySerde`, `ByteBufferSerde`, `BytesSerde` all in the apache code base which does very similar functionalities, and felt that maybe we can just keep the first and last one, while let users to get their own implementation if they really want `ByteBuffer` for serialization and deserialization.